docs/standard/native-interop/interop-guidance.md

```diff
@@ -0,0 +1,247 @@
+
+The guidance in this section applies to all interop scenarios.
+
+- **✔️ DO** use the same naming and capitalization for your methods and parameters as the native method you want to call.
+- **✔️ CONSIDER** using the same naming and capitalization for constant values.
+- **✔️ DO** use .NET types that map closest to the native type. For example, in C#, use `uint` when the native type is `unsigned int`.
+- **✔️ DO** only use `[In]` and `[Out]` attributes when the behavior you want differs from the default behavior.
+- **✔️ CONSIDER** using <xref:System.Buffers.ArrayPool%601?displayProperty=nameWithType> to pool your native array buffers.
+- **✔️ CONSIDER** wrapping your P/Invoke declarations in a class with the same name and capitalization as your native library.
+  - This allows your `[DllImport]` attributes to use the C# `nameof` language feature to pass in the name of the native library and ensure that you did not misspell the name of the native library.
+
+## [DllImport](xref:System.Runtime.InteropServices.DllImportAttribute) attribute settings
+
+| Setting | Default | Recommendation | Details |
+|---------|---------|----------------|---------|
+| [`PreserveSig`](xref:System.Runtime.InteropServices.DllImportAttribute.PreserveSig)   | `true` |  keep default  | When this is explicitly set to false, failed HRESULT return values will be turned into Exceptions (and the return value in the definition becomes null as a result).|
+| [`SetLastError`](xref:System.Runtime.InteropServices.DllImportAttribute.SetLastError) | `false`  | depends on the API  | Set this to true if the API uses GetLastError and use Marshal.GetLastWin32Error to get the value. If the API sets a condition that says it has an error, get the error before making other calls to avoid inadvertently having it overwritten.|
+| [`CharSet`](xref:System.Runtime.InteropServices.DllImportAttribute.CharSet) | `CharSet.None`, which falls back to `CharSet.Ansi` behavior  | Explicitly  use `CharSet.Unicode` or `CharSet.Ansi` when strings or characters are present in the definition | This specifies marshalling behavior of strings and what `ExactSpelling` does when `false`. Note that `CharSet.Ansi` is actually UTF8 on Unix. _Most_ of the time Windows uses Unicode while Unix uses UTF8. See more information on the [documentation on charsets](./charset.md). |
+| [`ExactSpelling`](xref:System.Runtime.InteropServices.DllImportAttribute.ExactSpelling) | `false` | `true`             | Set this to true and gain a slight perf benefit as the runtime will not look for alternate function names with either an "A" or "W" suffix depending on the value of the `CharSet` setting ("A" for `CharSet.Ansi` and "W" for `CharSet.Unicode`). |
+
+## String parameters
+
+When the CharSet is Unicode or the argument is explicitly marked as `[MarshalAs(UnmanagedType.LPWSTR)]` _and_ the string is passed by value (not `ref` or `out`) the string will be be pinned and used directly by native code (rather than copied).
```

Comment:
```suggestion
When the CharSet is Unicode or the argument is explicitly marked as `[MarshalAs(UnmanagedType.LPWSTR)]` _and_ the string is passed by value (not `ref` or `out`), the string will be pinned and used directly by native code (rather than copied).
```